Source/Diagnostics/ReducedDiags/LoadBalanceCosts.cpp

```diff
@@ -0,0 +1,194 @@
+            ofs << "#";
+            ofs << "[1]step()";
+            ofs << m_sep;
+            ofs << "[2]time(s)";
+
+            for (int boxNumber=0; boxNumber<m_nBoxesMax; ++boxNumber)
+            {
+                ofs << m_sep;
+                ofs << "[" + std::to_string(3 + m_nDataFields*boxNumber) + "]";
+                ofs << "cost_box_"+std::to_string(boxNumber)+"()";
+                ofs << m_sep;
+                ofs << "[" + std::to_string(4 + m_nDataFields*boxNumber) + "]";
+                ofs << "proc_box_"+std::to_string(boxNumber)+"()";
+                ofs << m_sep;
+                ofs << "[" + std::to_string(5 + m_nDataFields*boxNumber) + "]";
+                ofs << "lev_box_"+std::to_string(boxNumber)+"()";
+                ofs << m_sep;
+                ofs << "[" + std::to_string(6 + m_nDataFields*boxNumber) + "]";
+                ofs << "i_low_box_"+std::to_string(boxNumber)+"()";
+                ofs << m_sep;
+                ofs << "[" + std::to_string(7 + m_nDataFields*boxNumber) + "]";
+                ofs << "j_low_box_"+std::to_string(boxNumber)+"()";
+                ofs << m_sep;
+                ofs << "[" + std::to_string(8 + m_nDataFields*boxNumber) + "]";
+                ofs << "k_low_box_"+std::to_string(boxNumber)+"()";
+            }
+            ofs << std::endl;
+
+            // open the data-containing file
+            std::string fileDataName = m_path + m_rd_name + "." + m_extension;
+            std::ifstream ifs(fileDataName, std::ifstream::in);
+
+            // Fill in the tmp costs file with data, padded with NaNs
+            for (std::string lineIn; std::getline(ifs, lineIn);)
+            {
+                // count the elements in the input line
+                int cnt = 0;
+                std::stringstream ss(lineIn);
+                std::string token;
+
+                while (std::getline(ss, token, m_sep[0]))
+                {
+                    cnt += 1;
+                    if (ss.peek() == m_sep[0]) ss.ignore();
+                }
+
+                // 2 columns for step, time; then nBoxes*nDatafields columns for data;
+                // then fill the remaining columns (i.e., up to 2 + m_nBoxesMax*m_nDataFields)
+                // with NaN, so the array is not jagged
+                ofs << lineIn;
+                for (int i=0; i<(m_nBoxesMax*m_nDataFields - (cnt - 2)); ++i)
+                {
+                    ofs << m_sep << "NaN";
+                }
+                ofs << std::endl;
+            }
+
+            // close files
+            ifs.close();
+            ofs.close();
+
+            // remove the original, rename tmp file
+            std::remove(fileDataName.c_str());
+            std::rename(fileTmpName.c_str(), fileDataName.c_str());
+
+        } // if (step == (warpx.maxStep() - 1)) ...
+    } // if (ParallelDescriptor::IOProcessor()) ...
+} // end ReducedDiags::WriteToFile
```

Comment:
Same as above (and it also applies in other places).

Reply:
I've removed all comments like this in the current PR.  But I think these are common throughout the code, there are certainly comments like this in e.g. `FieldEnergy.cpp` and `ParticleEnergy.cpp`.